src/back-end/lib/resources/affiliation.ts

```diff
@@ -58,63 +56,58 @@ const resource: Resource = {
 
   create(connection) {
     return {
-      async parseRequestBody(request) {
-        return request.body.tag === 'json' ? request.body.value : {};
+      async parseRequestBody(request): Promise<CreateRequestBody> {
```

Comment:
You shouldn't need to type the return value of this function, as it should be inferred by the `Resource` type on line 31.

Reply:
So I thought that was the case, but some of the functions were not showing compile-time errors unless I explicitly put this in.  I'll take another look though, maybe I was not doing something correctly.

Reply:
Yeah, I confirmed that if I remove the typing of the return value, it will not type check the return value properly.  This was the source of the my previous errors that resulted in the need for DM-169 (Review CRUD resources to ensure the result from parseRequestBody matches the `Create/UpdateRequestBody` types in shared.).  Let me know if I am missing something, or if there is a better way to do this - seems like a compiler limitation with TypeScript.

Reply:
I think the reason the types were still passing were because `JsonRequestBody`'s values are types as `any` by default.

Reply:
I'll try changing that to `unknown` to see if that solves the problem.

Reply:
I ended up changing the `JsonRequestBody` type to use a new `JsonValue` type I defined. I couldn't do this until TypeScript 3.7, because the language didn't support recursive type aliases. Have a look at https://github.com/bcgov/digital_marketplace/pull/32/commits/6d94653948176d5e6ff0db3fb6e72bccb24504c9 if you are interested.

I have also removed the unnecessary `parseRequestBody` type signatures.

Reply:
Ah, nice thanks for that.  I can clean up the rest of the unnecessary return types.